AdyenComponents/Affirm/AffirmComponent.swift

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Adyen
+import UIKit
+
+/// A component that provides a form for Affirm payment.
+public final class AffirmComponent: AbstractPersonalInformationComponent, Observer {
+    
+    private enum ViewIdentifier {
+        static let billingAddress = "billingAddressItem"
+        static let deliveryAddress = "deliveryAddressItem"
+        static let deliveryAddressToggle = "deliveryAddressToggleItem"
+    }
+    
+    // MARK: - Items
+    
+    /// :nodoc:
+    private let personalDetailsHeaderItem: FormLabelItem
+    
+    /// :nodoc:
+    internal let deliveryAddressToggleItem: FormToggleItem
+    
+    /// :nodoc:
+    internal let deliveryAddressItem: FormAddressItem
+    
+    // MARK: - Initializers
+    
+    /// Initializes the Affirm component.
+    /// - Parameters:
+    ///   - paymentMethod: The Affirm payment method.
+    ///   - apiContext: The component's API context.
+    ///   - style: The component's style.
+    public init(paymentMethod: PaymentMethod,
+                apiContext: APIContext,
+                style: FormComponentStyle) {
+        personalDetailsHeaderItem = FormLabelItem(text: "", style: style.sectionHeader)
+        deliveryAddressToggleItem = FormToggleItem(style: style.toggle)
+        deliveryAddressItem = FormAddressItem(initialCountry: Locale.current.regionCode ?? "US", style: style.addressStyle)
+        
+        let fields: [PersonalInformation] = [
+            .custom(CustomFormItemInjector(item: personalDetailsHeaderItem.addingDefaultMargins())),
+            .firstName,
+            .lastName,
+            .email,
+            .phone,
+            .address,
```

Comment:
I am wondering if you can use two `.address(title)` items instead of making a `CustomFormItemInjector ` with another `FormAddressItem `, but this can be in a different PR

Reply:
Actually it's a good idea. We can update the `addressItemInjector` on the `AbstractPersonalInformationComponent` so we pass the title